Mage.Sets/src/mage/cards/k/KjeldoranEliteGuard.java

```diff
@@ -0,0 +1,128 @@
+package mage.cards.k;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.DelayedTriggeredAbility;
+import mage.abilities.condition.common.IsPhaseCondition;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.decorator.ConditionalActivatedAbility;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.SacrificeTargetEffect;
+import mage.abilities.effects.common.continuous.BoostTargetEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class KjeldoranEliteGuard extends CardImpl {
+
+    public KjeldoranEliteGuard(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{W}");
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Target creature gets +2/+2 until end of turn.
+        // When that creature leaves the battlefield this turn, sacrifice Kjeldoran Elite Guard.
+        // Activate only during combat.
+        Ability ability = new ConditionalActivatedAbility(
+                Zone.BATTLEFIELD,
+                new KjeldoranEliteGuardEffect(),
+                new TapSourceCost(),
+                new IsPhaseCondition(TurnPhase.COMBAT, false));
+        ability.addTarget(new TargetControlledCreaturePermanent());
```

Comment:
This should just be `TargetCreaturePermanent`

Reply:
What is the difference? I couldn't figure it out just based on class names

Reply:
`TargetControlledCreaturePermanent` is a creature you control whereas `TargetCreaturePermanent` is any creature